api/pkg/service/resource/resource_http_test.go

```diff
@@ -641,3 +641,37 @@ func TestByID_Http_ErrorCase(t *testing.T) {
 		assert.Equal(t, "not-found", err.Name)
 	})
 }
+
+func TestDeprecationByVersionID_Http(t *testing.T) {
```

Comment:
Should we add test when the resource is not deprecated ??

Reply:
It's not necessary and that case would be cover in `TestByVersionID_Http` test